### ISSUE TITLE
[bug]: The Asterisk REST Interface is Currently Disabled

### ISSUE THREAD

**Issue by LeslieDesser** [FreePBX/issue-tracker#132]
### FreePBX Version

FreePBX 17

### Issue Description

I was not sure how to list this but have chosen Bug as I think that is what it is.

This is a brand new install from a few days ago.  I noticed when looking at Asterisk Info, this message was showing in red in the Channel and some other sections.   I got no response from the Community and the message does not come up with any hits in Google.  After about a couple of hours of messing about I discovered ARI = Asterisk REST Interface.  ARI returned some hits which eventually sent me to Advanced Setting to enable this option.

So first bug:  Why can't the message include some hint as to where to go?  

Second bug:  If the ARI must be enabled for FreePBX to work as it should, then why is it off by default?

So I enabled it.  The message has now changed to `The Asterisk REST Interface is not able to connect please check configuration in advanced settings`.

I have checked, have googled, have manually created an ARI user but nothing changes.

PS: Asterisk Logfiles seems to have gone AWOL.

### Operating Environment

Latest Debian 12, latest F17 script.  Same details as #128

**Comment by dolesec** [comment]
@LeslieDesser its not required in general however if you want to use things like the Sangoma Desktop Phone you will be warned within userman that it is a requirement 

I toggled this on and off and watched what was reported by asterisk info ... when disabled in advanced settings asterisk info would indeed report it disabled; after enabling there was no message similar to what you are reporting ... and there is no need to create a manual user 

does your config look similar to below?

![Screenshot 2024-05-05 at 4 55 37 PM](https://github.com/FreePBX/issue-tracker/assets/2153999/65dcd16c-4de5-4b3d-b0bd-e5ade54a0107)

review this older post as well ->  https://community.freepbx.org/t/the-asterisk-rest-interface-is-not-able-to-connect-please-check-configuration-in-advanced-settings/88522/6  (in particular the bit about the Asterisk Builtin mini-HTTP server HTTP Prefix field) 


btw - logfiles is now system logfiles
 
![Screenshot 2024-05-05 at 5 05 04 PM](https://github.com/FreePBX/issue-tracker/assets/2153999/65ebe764-c22e-4bbf-8b1d-666a5ea08b05)

**Comment by LeslieDesser** [comment]
My REST Interface settings are exactly like yours.  The mini-HTTP server is enabled.

I had seen the thread you pointed to - the Prefix is blank.  The User and password fields have been removed so that is not relevant.

Is there a log that may show up the reason for the failure?  I have checked a few without luck.
------------
As to System Log Files - thank you for the pointer.  But I wonder why the name has changed.  Surely it is not the System log file.  It is specifically the Asterisk log file.

**Comment by LeslieDesser** [comment]
OK - problem solved.  The thread you linked was the solution.

1. In Advanced settings: Set `Display Readonly Settings`  to `Yes` and also set `Override Readonly Settings`  to `Yes`
2. Look in settings `ARI Username`  and `ARI Password` for any non-alphanumeric characters and REMOVE.  I found `/+` as part of the name.
3. Submit and Apply Config.
4. Remember to turn off the settings made in step 1.

Problem solved.

So this is really an old bug - the system is still automatically generating random user/password values which are invalid.

Also, if it is important that by default ARI should be turned off, then wherever this results in a failure (like in Asterisk info which I would think is a core option) then clear instructions on how to turn it on should be included.)

@dolesec Thank you for your prompt responses.

**Comment by kapilgupta01** [comment]
Hi @LeslieDesser Yes its our default settings right now to keep it disabled. 

Just to make sure user is enabling it when its required not leaving it on unattended. 

thanks

**Comment by nicolasmtz** [comment]
I have the same problem mentioned here.  Here are some facts and steps I've followed:

1. Issue happening after a fresh install with Debian 12.7 and the latest FreePBX code as of October 12, 2024 (CST).
2. Asterisk REST Interface is enabled in Advanced Settings.
3. No special characters in ARI username and password.
4. HTTP Prefix is blank in Asterisk Builtin mini-HTTP server (Advanced Settings).
5. Removed and reinstalled arimanager and asteriskinfo modules.

Anything else that we can check so we can activate REST Interface?
